### DISCUSSION TITLE
Vanilla dataclass w/ UUID as ResponseModel crashes with missing serializer

### DISCUSSION THREAD

**Discussion by haf** [fastapi/fastapi#8493] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from uuid import UUID, uuid4
from dataclasses import dataclass

app = FastAPI()

@dataclass
class A:
    value: UUID

@app.get("/")
def example():
  return A(value=uuid4())
```

### Description

```bash
cat x.py <<EOF
from fastapi import FastAPI
from uuid import UUID, uuid4
from dataclasses import dataclass

app = FastAPI()

@dataclass
class A:
    value: UUID

@app.get("/")
def example():
  return A(value=uuid4())
EOF
pip install uvicorn fastapi
uvicorn x:app
curl http://127.0.0.1:8000
```

### Operating System

macOS

### FastAPI Version

0.75.2

### Python Version

3.9.10

### Additional Context

Since FastAPI supports UUID input, in parameters and such, it should support returning them in dataclasses.

- https://fastapi.tiangolo.com/tutorial/extra-data-types/
- https://fastapi.tiangolo.com/advanced/dataclasses/

I can't use a pydantic base class because that base class doesn't properly pickle with `cloudpickle` that we use as a read-through cache for machine learning models.

**Comment by gregbrowndev** [comment]
This is also true for `datetime`, enum classes, and probably more. E.g.

```py
@dataclass
class A:
    created: datetime
```

and

```py
class Status(enum.Enum):
    OK = "OK"
    BAD = "BAD"


@dataclass
class A:
    status: Status
```

**Comment by YuriiMotov** [comment]
There is no response model specified in your code examples.

Anyway, this works fine in current version
